pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -389,6 +387,9 @@ func (cd *clusterDeploy) deployAgent(cluster *apimgmtv3.Cluster) error {
 
 	cluster.Status.AppliedAgentEnvVars = append(settings.DefaultAgentSettingsAsEnvVars(), cluster.Spec.AgentEnvVars...)
 
+	cluster.Status.AppliedClusterAgentDeploymentCustomization = cluster.Spec.ClusterAgentDeploymentCustomization
+	cluster.Status.AppliedFleetAgentDeploymentCustomization = cluster.Spec.FleetAgentDeploymentCustomization
```

Comment:
Now that I look into it, do we actually need `AppliedFleetAgentDeploymentCustomization`?

Even if we do, I'm not sure why we would be setting it on _`cattle-cluster-agent`_ (and not `fleet-agent` deployment)? And since we don't directly handle `fleet-agent` deployment that another point to question existence of `AppliedFleetAgentDeploymentCustomization`.